File: weekly/349/q1.cc
class Solution {
public:
    int findNonMinOrMax(vector<int>& nums) {
        if (nums.size() <= 2) return -1;
        sort(nums.begin(), nums.end());
        int n = nums.size(), i;
        for (i = 1; i < n; ++i) {
            if (nums[i] == nums.back()) {
                i = -1;
                break;
            }
            if (nums[i] != nums.front()) {
                break;
            }
        }
        if (i == -1) { return -1; }
        return nums[i];
    }
};

File: weekly/349/go_a/q1.go
func findNonMinOrMax(nums []int) int {
    arr := make([]int, 3)
    PH := -114514
    for i := range arr {
        arr[i] = PH
    }
    for _, v := range nums {
        if arr[0] != PH && arr[1] != PH && arr[2] != PH {
            sort.Ints(arr)
            return arr[1]
        }
        if v != arr[0] && v != arr[1] && v != arr[2] {
            for i := range arr {
                if arr[i] == PH {
                    arr[i] = v
                    break
                }
            }
        }
    }

    if arr[0] != PH && arr[1] != PH && arr[2] != PH {
        sort.Ints(arr)
        return arr[1]
    }
    return -1
}

